server/controllers/profile.js

```diff
@@ -0,0 +1,118 @@
+const Profile = require("../models/Profile");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+exports.getProfile = asyncHandler(async (req, res, next) => {
+  const id = req.body.id;
```

Comment:
Once we've added the `protect` middleware, we can obtain the logged in user's id through the request object - `req.user.id` 

Note: It is also not advisable to send request body params in a GET request